executor/show.go

```diff
@@ -58,41 +58,11 @@ type ShowExec struct {
 	is infoschema.InfoSchema
 
 	forChunk bool
```

Comment:
after removing `Next`, `forChunk` will always be true, which should also be removed.

Reply:
Why did I remove `forChunk`
When I do a unit test, I get a pannic:
```
PANIC: ddl_test.go:45: testSuite.TestCreateTable

... Panic: runtime error: invalid memory address or nil pointer dereference (PC=0x45BCDA)

/home/zhoudazhuang/usr/local/go/go1.9.4/src/runtime/panic.go:491
  In gopanic
/home/zhoudazhuang/usr/local/go/go1.9.4/src/runtime/panic.go:63
  In panicmem
/home/zhoudazhuang/usr/local/go/go1.9.4/src/runtime/signal_unix.go:367
  In sigpanic
/home/zhoudazhuang/class100/src/github.com/pingcap/tidb/util/chunk/chunk.go:251
  In ShowExec.appendRow
Github.com/pingcap/tidb/executor/_test/_obj_test/show.go:1027
  In ShowExec.appendRow
Github.com/pingcap/tidb/executor/_test/_obj_test/show.go:398
  In ShowExec.fetchShowColumns
Github.com/pingcap/tidb/executor/_test/_obj_test/show.go:398
  In ShowExec.fetchShowColumns
Github.com/pingcap/tidb/executor/_test/_obj_test/show.go:158
  In ShowExec.fetchAll
Github.com/pingcap/tidb/executor/_test/_obj_test/show.go:69
  In ShowExec.Next
Github.com/pingcap/tidb/executor/_test/_obj_test/adapter.go:92
  In recordSet.Next
Ddl_test.go:68
  In testSuite.TestCreateTable
/home/zhoudazhuang/usr/local/go/go1.9.4/src/reflect/value.go:302
  In Value.Call
/home/zhoudazhuang/class100/src/github.com/pingcap/tidb/vendor/github.com/pingcap/check/check.go:798
  In suiteRunner.forkTest.func1
/home/zhoudazhuang/class100/src/github.com/pingcap/tidb/vendor/github.com/pingcap/check/check.go:692
  In suiteRunner.forkCall.func1
/home/zhoudazhuang/usr/local/go/go1.9.4/src/runtime/asm_amd64.s:2337
  In goexit

```

Just this code at ddl_test:
```
//{&AlterTableStmt{Table: &TableName{}, Specs: []*AlterTableSpec{alterTableSpec}}, 0, 0},
```
I commented on this code, but it didn't work. How should I fix it?

Reply:
`forChunk` is always true, it's redundant, it was introduced to distinguish `Next` and `NextChunk`, we should remove this variable since `Next` is removed.

Reply:
you should rewrite`TestCreateTable` to not call `recordSet.Next`, call `recordSet.NextChunk` instead.

Reply:
sorry,but I can not find `TestCreateTable` to not call `recordSet.Next` in this pro

Reply:
我觉得应该不是调用`Next`的问题，因为我删除了`Next`方法，单元测试是可以通过的。直到我删除了`forChunk`,这个时候单元测试就不通过了

Reply:
maybe you can investigate the usage of `forChunk` in function `forChunk` and do some modification.

Reply:
I got it, Please PTAL